pkg/ddevapp/apptypes.go

```diff
@@ -127,3 +131,14 @@ func (app *DdevApp) SetApptypeSettingsPaths() {
 		appFuncs.apptypeSettingsPaths(app)
 	}
 }
+
+// DetectAppType calls each apptype's detector until it finds a match,
+// or returns 'php' as a last resort.
+func (app *DdevApp) DetectAppType() string {
+	for appName, appFuncs := range appTypeMatrix {
+		if appFuncs.appTypeDetect != nil && appFuncs.appTypeDetect(app) {
+			return appName
+		}
+	}
+	return "php"
```

Comment:
IMO, we should not hardcode this. Put `php` at the end of `appTypeMatrix`, and then have an actual detector function for PHP that returns true if there are any PHP files in the app root. It'll be last in line behind the other PHP app types. That way, if we ever get around to adding Ruby or Python environments, we can do something similar for fallback detection.

Reply:
@rfay let me know that the ordering of maps in Go is indeterminate, so this is not quite as easy as I thought it would be. Given that support for other languages is likely a while out still, this is fine as is :)